addons/mail/controllers/discuss.py

```diff
@@ -107,3 +107,7 @@ def message_get_suggested_recipients(self, model, res_ids):
         except Exception:
             return {}
         return records._message_get_suggested_recipients()
+
+    @http.route('/mail/edit_message/<model("mail.message"):message>', type='json', auth='user')
+    def edit_message(self, message, body):
+        return request.env[message.model]._message_edit_content(message, body)
```

Comment:
Don't use slug, it will fail for guest because it will be done before the access check and the sudo giving access. Just have the id instead. Also I'm not sure yet if we want the id actually on the URL or just give it as post param.

Also the method should probably be forced to be POST only, I don't remember if it's automatic with type json or not, but requests changing stuff should always be POST to avoid security issues.